update acorn depencency, add acorn options for import/export

BenDiuguid opened the pull request:
As discussed in #21 I have updated the dependencies, and added some acorn options to allow for the parsing of `import`/`export` statements.

Let me know if you need me to make any changes!

MichaReiser commented on src/js/translate-visitor.ts:
```diff
@@ -7,15 +7,17 @@ const n = types.namedTypes;
         onToken: this.tokens,
-        ranges: true
+        ranges: true,
+        sourceType: "module",
```
I don't think we should enable this by default. The problem is that this enables strict mode for all files. So, this might fail to parse in case someone is still using commonjs with any feature that is not supported by ES6.

So I think it would be best if we allow the user to pass options to the js-loader (like you can to other loaders) and merge them with the default options defined by the webpack-plugin....

MichaReiser commented on src/js/translate-visitor.ts:
```diff
@@ -7,15 +7,17 @@ const n = types.namedTypes;
-        ranges: true
+        ranges: true,
+        sourceType: "module",
+        allowImportExportEverywhere: true
```
The same applies for this option...

BenDiuguid commented:
Makes sense! Would you like me to Bump up the minor version in the `package.json` as well?

BenDiuguid commented:
I updated the README.md with a little bit of documentation on working in the options with webpack v1 and webpack v2, and I have tested it locally in a webpack v2 project that I'm working on. I also updated the example and tested it with webpack v1.

I guess I need to update the tests to improve the coverage slightly 😃

BenDiuguid commented:
Tests updated, let me know what you think!

MichaReiser commented on src/js/js-loader.ts:
```diff
@@ -69,4 +70,14 @@ function isExcludedResource(resource: string): boolean {
 
-export = jsLoader;
\ No newline at end of file
+function parseOptions(version: number, query: any) {
```
Why are you not using [loader-utils](https://github.com/webpack/loader-utils) to parse the query. As I remember, the query can also be a string, in which case this will fails?

MichaReiser commented on src/js/js-loader.ts:
```diff
@@ -32,9 +32,10 @@ function jsLoader(source: string, sourceMaps: any): void {
     loader.pruneTranslations(loader.resource);
 
-    const visitor = new TranslateVisitor(loader);
+    const visitor = new TranslateVisitor(loader, options);
```
Can we just pass `options.parserOptions` instead. Makes it easier to separate the options in case we add new ones in the future.

MichaReiser commented on src/js/translate-visitor.ts:
```diff
@@ -19,9 +19,13 @@ export default class TranslateVisitor extends types.PathVisitor implements  type
+    constructor(private loader: TranslateLoaderContext, options: any) {
         super();
 
+        options = options || {};
```
You can use `Object.assign` here to merge the user options with our default options.

```js
this.options = Object.assign({}, options, this.options);
```

BenDiuguid replied:
The current version of typescript/typings is causing issues with Object.assign and object rest spread, so I ended up using an npm package which does the same thing [object-assign](https://www.npmjs.com/package/object-assign)

MichaReiser commented on README.MD:
```diff
@@ -35,7 +36,30 @@ module: {
+            loader: WebPackAngularTranslate.jsLoader(),
+            query: {
+                sourceType: 'script', // acorn option default
+                allowImportExportEverywhere: false // acorn option default
```
I don't think that this option is needed in all cases. Normally the `sourceType` option should be sufficient. The latter is used in cases where imports are used inside of functions, and therefore, is not conform with the current spec (I believe)

BenDiuguid replied:
Either way that was just the default option, but I needed to use it for some reason. But I'll remove this from the example either way. `sourceType: 'script'` is also the default. The defaults are listed [here](https://github.com/ternjs/acorn/blob/master/src/options.js)

BenDiuguid commented:
I have time to tackle these changes tomorrow!

And I agree with all your comments! I didn't know about `loader-utils` so that's really cool!

And I tried object assign and object spread, but I wasn't sure if the version of typescript supported it?

I have a lot of is experience, but very little typescript experience :smile:

BenDiuguid commented:
I addressed your comments, and things should be good to go 👍

MichaReiser commented:
Thx for your pull request. I made some smaller adjustments, but is now merged and released!

BenDiuguid commented:
Awesome! Thanks so much!
